Assets/MixedRealityToolkit/Extensions/CameraExtensions.cs

```diff
@@ -36,10 +37,73 @@ public static bool IsInFOV(this Camera camera, Vector3 position)
             Vector3 screenPoint = camera.WorldToViewportPoint(position);
 
             return screenPoint.z >= camera.nearClipPlane && screenPoint.z <= camera.farClipPlane
-                && screenPoint.x >= 0 && screenPoint.x <= 1 
+                && screenPoint.x >= 0 && screenPoint.x <= 1
                 && screenPoint.y >= 0 && screenPoint.y <= 1;
         }
 
+
+        // Help to clear caches when new frame runs
+        static private int lastCalculatedFrame = -1;
+        // Map from grabbable => is the grabbable in FOV for this frame. Cleared every frame
+        private static Dictionary<Collider, bool> colliderCache = new Dictionary<Collider, bool>();
+        // List of corners shared across all sphere pointer query instances --
+        // used to store list of corners for a bounds. Shared and static
+        // to avoid allocating memory each frame
+        private static List<Vector3> corners = new List<Vector3>();
+
+        /// <summary>
+        /// Returns true if a collider's bounds is within the camera FOV. 
+        /// Utilizes a cache to test if this collider has been seen before and returns current frame's calculated result.
+        /// </summary>
+        /// <param name="myCollider">The collider to test</param>
+        public static bool IsInFOVConeCached(this Camera cam,
+            Collider myCollider)
+        {
+            if (lastCalculatedFrame != Time.frameCount)
+            {
+                colliderCache.Clear();
+                lastCalculatedFrame = Time.frameCount;
+            }
+
+            if (colliderCache.TryGetValue(myCollider, out bool result))
```

Comment:
minor nit: make else if (colliderCache.Try) 
If we clear the cache, then we know there is no point in trying to test access

Reply:
Done